pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -0,0 +1,254 @@
+		Message: "Job finished successfully",
+	}
+	if ph == corev1.PodFailed {
+		condition.Message = "Job failed"
+	}
+
+	return condition, ph == corev1.PodSucceeded || ph == corev1.PodFailed
+}
+
+// PodSets will build workload podSets corresponding to the job.
+func (p *Pod) PodSets() []kueue.PodSet {
+	return []kueue.PodSet{
+		{
+			Name:  kueue.DefaultPodSetName,
+			Count: 1,
+			Template: corev1.PodTemplateSpec{
+				Spec: *p.Spec.DeepCopy(),
+			},
+		},
+	}
+}
+
+// PriorityClass returns the job's priority class name.
+func (p *Pod) PriorityClass() string {
+	return p.Spec.PriorityClassName
+}
+
+// IsActive returns true if there are any running pods.
+func (p *Pod) IsActive() bool {
+	return p.Status.Phase == corev1.PodRunning
+}
+
+// PodsReady instructs whether job derived pods are all ready now.
+func (p *Pod) PodsReady() bool {
+	for i := range p.Status.Conditions {
+		c := &p.Status.Conditions[i]
+		if c.Type == corev1.PodReady {
+			return c.Status == corev1.ConditionTrue
+		}
+	}
+	return false
+}
+
+// GVK returns GVK (Group Version Kind) for the job.
+func (p *Pod) GVK() schema.GroupVersionKind {
+	return gvk
+}
+
+func (p *Pod) Stop(ctx context.Context, c client.Client, _ []jobframework.PodSetInfo, eventMsg string) (bool, error) {
+	// The podset info is not relevant here, since this should mark the pod's end of life
+	pCopy := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:       p.UID,
+			Name:      p.Name,
+			Namespace: p.Namespace,
+		},
+		TypeMeta: p.TypeMeta,
+		Status: corev1.PodStatus{
+			Conditions: []corev1.PodCondition{
+				{
+					Type:   ConditionTypeTerminationTarget,
+					Status: corev1.ConditionTrue,
+					LastTransitionTime: metav1.Time{
+						Time: time.Time{},
+					},
+					Reason:  "StoppedByKueue",
+					Message: eventMsg,
+				},
+			},
+		},
+	}
+	err := c.Status().Patch(ctx, pCopy, client.Apply, client.FieldOwner(constants.KueueName))
+	if err == nil {
+		err = c.Delete(ctx, p.Object())
+	}
+	if err == nil && apierrors.IsNotFound(err) {
+		return true, nil
+	}
+	return false, err
+}
+
+func SetupIndexes(ctx context.Context, indexer client.FieldIndexer) error {
+	return jobframework.SetupWorkloadOwnerIndex(ctx, indexer, gvk)
+}
+
+func (p *Pod) Finalize(ctx context.Context, c client.Client) error {
+	pod := (*corev1.Pod)(p)
+
+	if controllerutil.RemoveFinalizer(pod, PodFinalizer) {
+		if err := c.Update(ctx, pod); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *Pod) Skip() (bool, error) {
+	// Skip pod reconciliation, if managed label is not set
+	if v, ok := p.GetLabels()[ManagedLabelKey]; !ok || v != ManagedLabelValue {
+		return true, nil
+	}
+
+	if IsPodOwnerManagedByQueue(p) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func IsPodOwnerManagedByQueue(p *Pod) bool {
+	if owner := metav1.GetControllerOf(p); owner != nil {
+		return jobframework.IsOwnerManagedByKueue(owner) || (owner.Kind == "RayCluster" && strings.HasPrefix(owner.APIVersion, "ray.io/v1alpha1"))
```

Comment:
also MPIJob and all the kubeflow jobs. Maybe we can iterate over all integrations and call the same function?
And then RayCluster is special.